File: src/main/java/com/vjiki/music/mapper/PlaylistMapper.java
package com.vjiki.music.mapper;

import com.vjiki.music.dto.PlaylistResponse;
import com.vjiki.music.entity.Playlist;

public final class PlaylistMapper {

    private PlaylistMapper() {
    }

    public static PlaylistResponse toResponse(Playlist playlist) {
        return PlaylistResponse.builder()
                .id(playlist.getId())
                .userId(playlist.getUser() == null ? null : playlist.getUser().getId())
                .userName(playlist.getUser() == null ? null : playlist.getUser().getEmail())
                .userNickname(playlist.getUser() == null ? null : playlist.getUser().getNickname())
                .name(playlist.getName())
                .description(playlist.getDescription())
                .coverUrl(playlist.getCoverUrl())
                .type(playlist.getType())
                .isPublic(playlist.getIsPublic())
                .createdAt(playlist.getCreatedAt())
                .modifiedAt(playlist.getModifiedAt())
                .build();
    }
}

File: src/main/kotlin/com/vjiki/music/mapper/PlaylistMapper.kt
package com.vjiki.music.mapper

import com.vjiki.music.dto.PlaylistResponse
import com.vjiki.music.entity.Playlist
import org.springframework.stereotype.Component

@Component
object PlaylistMapper {
    fun Playlist.toResponse(): PlaylistResponse {
        return PlaylistResponse(
            id = id,
            userId = user.id,
            userName = user.email,
            userNickname = user.nickname,
            name = name,
            description = description,
            coverUrl = coverUrl,
            type = type,
            isPublic = isPublic,
            createdAt = createdAt,
            modifiedAt = modifiedAt
        )
    }
}

